### DISCUSSION TITLE
No audio when using systemctl start, but works fine when launched manually

### DISCUSSION THREAD

**Discussion by narFnarF** [mikebrady/shairport-sync#1476] [Q&A] [answered]
Hello

I'm a long time user of Shairport-sync on my Raspberry Pi 2B. This week, I completely reinstalled the OS on it from a fresh image to upgrade from Stretch to Buster.

Now I have a weird problem:
When I launch Shairport manually by typing `shairport-sync`, everything works. I can connect my phone and get music playing. 

But when I launch Shairport using `sudo systemctl restart shairport-sync.service`, I don't get any audio from my speakers. But I know it's running fine by looking at `service shairport-sync status` and I also see it on my iPhone. I can connect my phone to it, but then I get no audio.

If I manually type `sudo shairport-sync`, I get the same problem (connects but no audio)

What's wrong with my installation? How can I troubleshoot?

### What I tried so far:
- Installing shairport by using `sudo apt install shairport-sync`. This installs shairport-sync/oldstable,now 3.2.2-1+b2 armhf. As described above, it works when launched manually, but not via `systemctl start`
- Use `raspi-config` to set audio output to Headphones
- Check `alsamixer` to set the volume to maximum
- Doing a clean manual install from these instructions: https://github.com/mikebrady/shairport-sync/blob/master/INSTALL.md. Same behaviour.

### Device used:
Host: Raspberry Pi 2B, running Raspbian _Lite_ (Buster)
Streaming device: iPhone 6S, latest iOS version
Version : 3.3.9-OpenSSL-Avahi-ALSA-soxr-sysconfdir:/etc

### Setup
Speakers are plugged in the headphone jack of the Raspberry Pi 2B.

**Comment by narFnarF** [comment]
I guess this is the problem?
```
$ shairport-sync -d
shairport-sync was built without libdaemon, so does not support daemonisation using the -d, --daemon, -j or --justDaemoniseNoPIDFile options
```
but how is that possible?

I tried to reinstall the packaged version with `sudo apt install shairport-sync` and it launches correct with `sudo shairport-sync -d`.

One thing that I noticed: Launching manually with `shairport-sync` works but not with `sudo shairport-sync` (no audio)

**Comment by mikebrady** [comment]
> but how is that possible?

Thanks. This is possible because `systemd` can do its own daemonisation and doesn't need `libdaemon`.

My guess is that problem is that the startup script is faulty in some way (I vaguely recall something about it in older issues) , so I'd suggest that you manually remove it and build and reinstall Shairport Sync yourself.

**Comment by narFnarF** [comment, marked as answer]
For anyone with the same issue, I managed to fix this by starting over from the beginning _again_. Somehow this time worked.

So basically:
1. Uninstall the apt package
2. Follow the instructions on how to remove old packages and old startup script: ~~https://github.com/mikebrady/shairport-sync/blob/master/INSTALL.md~~ → https://github.com/mikebrady/shairport-sync/blob/master/BUILD.md
3. Reboot
4. Continue following the instruction on how to clone the git repo, build and install.

I don't understand why it didn't work the first few times but now it works and I'm happy. I really like Shairplay-sync, but I like it more when it works! 😜

**Reply by pilotkid** [reply]
[INSTALL.md](https://github.com/mikebrady/shairport-sync/blob/master/INSTALL.md) no longer exists.

**Reply by mikebrady** [reply]
Thanks @pilotkid. The installation guide was really a building guide 🙄. It has been moved to [BUILD.md](https://github.com/mikebrady/shairport-sync/blob/master/BUILD.md).
